### PR TITLE
Fixed handling of missing dependencies (fixes #150)

### LINKED ISSUE #150: Add support for missing rosinstall dependencies
See #126, #131, #127, #128. 

Possible approaches:

1. Each bug provides its own `deps.rosinstall.diff` patch file, which is applied to `deps.rosinstall` to obtain a complete list of dependencies. Has a number of cons: (1) tedious to determine appropriate versions, especially for recursive cases, (2) requires that we add a no-op file to every bug directory to ensure that bugs build, (3) more complicated (and needlessly so) than simply giving a list of additional dependencies.
2. Each bug provides its own `extra.rosinstall` file, which provides a `rosintall` file containing all of the missing dependencies. Slightly nicer than (1) but still tedious and introduces additional files.
3. Update `time-machine` section in bug description file to contain a list that details the missing dependencies for a particular bug:
```
missing-dependencies:
  - tar:
    local-name: actionlib
    uri: https://github.com/ros-gbp/actionlib-release/archive/release/hydro/actionlib/1.10.3-0.tar.gz
    version: actionlib-release-release-hydro-actionlib-1.10.3-0
```
  The contents of the `missing-dependencies` section can then either be automatically written to `extra.rosinstall` or folded into `deps.rosinstall`.
4. Update `time-machine` section in bug description file to state the names of the missing dependencies:
```
missing-dependencies:
  - actionlib
```
The time machine helper script, `build-time-machine.py`, should query the time machine to determine the most appropriate release of each missing dependency (along with any recursive dependencies). The complete set of dependencies should be folded into `deps.rosinstall`.

### PR DESCRIPTION
**Pull request by ChrisTimperley**
`--deps-only` was passed to the time machine along with the name of the missing dependency, meaning that the missing dependency itself wasn't added to the `.rosinstall` file.

### CODE REVIEW

File: scripts/build-rosinstall.py [hunk 1]
```diff
@@ -40,6 +40,33 @@ def gh_issue_to_datetime(url_issue):
...
+                             check=True,
+                             stdout=subprocess.PIPE)
+        contents = res.stdout.decode('utf-8')
+    except subprocess.CalledProcessError as err:
+        logger.warning("time machine failed (return code: %d) for bug [%s]",
+                       err.returncode, fn_bug_desc)
+        return
+
+    # updated repository names
+    contents = contents.replace('geometry_experimental', 'geometry2')
```
Comment (gavanderhoorn) [inline scripts/build-rosinstall.py]: Can we do this unconditionally?

**Comment by gavanderhoorn** [comment]
Fixes #150.

**Comment by gavanderhoorn** [comment]
Nice approach to deal with #150 btw.

File: scripts/build-rosinstall.py (hunk 1 above)
Comment (ChrisTimperley) [reply scripts/build-rosinstall.py]: Is there any case when we wouldn't want to do this? I'm not sure.

Comment (gavanderhoorn) [reply scripts/build-rosinstall.py]: I don't think there is actually, thinking some more about it.

`geometry_experimental` (as in: the repository) doesn't exist any more. And it won't come back either.

So all references to that repository should be updated to point to its current name, which would be `geometry2`.

Could we perhaps print a notice that `build-rosinstall.py` has renamed entries in the rosinstall file?

(perhaps too difficult/involved) could it add a comment to the `.rosinstall` file?

Comment (ChrisTimperley) [reply scripts/build-rosinstall.py]: Sure. We could print a notice and add a comment to the top of the `.rosinstall` file.

**Comment by gavanderhoorn** [comment]
Something weird going on. Github claims that there are no changes in this PR?

**Comment by ChrisTimperley** [comment]
> Something weird going on. Github claims that there are no changes in this PR?

Urgh. I messed up by merging this PR into another PR (for a mavros bug) and then merging that second PR.

File: scripts/build-rosinstall.py (hunk 1 above)
Comment (ChrisTimperley) [reply scripts/build-rosinstall.py]: Done. Do you mind checking the wording of the message, @gavanderhoorn?

**Pull request merged**